src/material/button/button-lazy-ripple-renderer.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {DOCUMENT} from '@angular/common';
+import {
+  ANIMATION_MODULE_TYPE,
+  Inject,
+  Injectable,
+  NgZone,
+  OnDestroy,
+  Optional,
+} from '@angular/core';
+import {
+  MAT_RIPPLE_GLOBAL_OPTIONS,
+  RippleConfig,
+  RippleGlobalOptions,
+  RippleRenderer,
+  RippleTarget,
+} from '../core';
+import {Platform} from '@angular/cdk/platform';
+
+@Injectable({providedIn: 'root'})
+export class MatButtonRippleLoader implements OnDestroy {
+  private _document: Document;
+
+  constructor(
+    private _platform: Platform,
+    private _ngZone: NgZone,
+    @Optional() @Inject(DOCUMENT) document: any,
+    @Optional() @Inject(ANIMATION_MODULE_TYPE) private animationMode?: string,
+    @Optional()
+    @Inject(MAT_RIPPLE_GLOBAL_OPTIONS)
+    private globalRippleOptions?: RippleGlobalOptions,
+  ) {
+    this._document = document;
+
+    this._ngZone.runOutsideAngular(() => {
+      const options = {passive: true, capture: true};
+      this._document.addEventListener('focus', this.onInteraction, options);
+      this._document.addEventListener('mouseenter', this.onInteraction, options);
+    });
+  }
+
+  ngOnDestroy() {
+    this._ngZone.runOutsideAngular(() => {
+      document.removeEventListener('focus', this.onInteraction);
```

Comment:
this._document? also I'm pretty sure you need to specify the same options you did when registering it